pagination/offsetpagination/paginator.go

```diff
@@ -0,0 +1,79 @@
+package offsetpagination
+
+import (
+	"net/http"
+	"strconv"
+)
+
+const (
+	defaultPageSize int = 100
+)
+
+type (
+	Paginator[T any] struct {
+		page                  int
+		pageSize, maxPageSize int
+	}
+	Result[T any] struct {
+		Items    []T  `json:"items"`
+		NextPage *int `json:"next_page,omitempty"`
```

Comment:
I personally would remove the pointer here, since `omitempty` works with default values too.